hera_pspec/uvpspec.py

```diff
@@ -80,44 +83,91 @@ def __init__(self):
+                            "norm", "norm_units", "taper", "cosmo", "beamfile",
+                            'folded', "store_cov"]
+        self._ndarrays = ["spw_array", "freq_array", "dly_array", "pol_array",
+                          "lst_1_array", "lst_avg_array", "time_avg_array",
+                          "lst_2_array", "time_1_array", "time_2_array",
+                          "blpair_array", "OmegaP", "OmegaPP", "beam_freqs",
+                          "bl_vecs", "bl_array", "telescope_location",
+                          "scalar_array", "labels", "label_1_array",
+                          "label_2_array"]
+
+        self._dicts = ["data_array", "wgt_array", "integration_array",
+                       "nsample_array", "cov_array"]
+        self._dicts_of_dicts = ["stats_array"]
 
         # define which attributes are considred meta data. Large attrs should be constructed as datasets
 
-        self._meta_dsets = ["lst_1_array", "lst_2_array", "time_1_array", "time_2_array", "blpair_array",
-                            "bl_vecs", "bl_array", 'lst_avg_array', 'time_avg_array', 'OmegaP', 'OmegaPP',
-                            "label_1_array", "label_2_array"]
-        self._meta_attrs = sorted(set(self._all_params) - set(self._dicts) - set(self._meta_dsets))
+        self._meta_dsets = ["lst_1_array", "lst_2_array", "time_1_array",
+                            "time_2_array", "blpair_array", "bl_vecs",
+                            "bl_array", "lst_avg_array", "time_avg_array",
+                            "OmegaP", "OmegaPP", "label_1_array",
+                            "label_2_array"]
+        self._meta_attrs = sorted(set(self._all_params) - set(self._dicts) - set(self._meta_dsets) - set(self._dicts_of_dicts))
         self._meta = sorted(set(self._meta_dsets).union(set(self._meta_attrs)))
 
         # check all params are covered
-        assert len(set(self._all_params) - set(self._dicts) - set(self._immutables) - set(self._ndarrays)) == 0
+        assert len(set(self._all_params) - set(self._dicts) - set(self._immutables) - set(self._ndarrays) - set(self._dicts_of_dicts)) == 0
 
         # Default parameter values
         self.folded = False
 
+    def get_cov(self,key,*args):
+        """
+        Slice into covariance array with a specified data key in the format
+        (spw, ((ant1, ant2),(ant3, ant4)), pol)
+
+        or
+
+        (spw, blpair-integer, pol)
+
+        where spw is the spectral window integer, ant1 etc. arge integers,
+        and pol is either a polarization string (ex. 'XX') or integer (ex. -5).
+
+        Parameters
+        ----------
+        key: tuple
+            Baseline-pair key
+
+        Returns
+        -------
+        data : complex ndarray
+            Shape (Ntimes, Ndlys, Ndlys)
+        """
+        spw, blpairts, pol = self.key_to_indices(key, *args)
+        # Need to deal with folded data!
+        # if data has been folded, return only positive delays
```

Comment:
this comment is out of place: this return statement doesn't deal w/ folded data. we should add a line that does properly deal w/ folded data

Reply:
I've added covariance matrix folding and get_cov now deals with the folded case so this comment is now appropriate :)

Reply:
great thanks!

Reply:
@nkern I kind of skirted over the folded data line (and added this comment to the code) because I was unsure of how to deal with folded data. Currently stats array isn't touched by grouping.fold_spectra, so it wouldn't make sense to only use the second half of the errors, as is done in get_data if self.folded is True. What to do to grouping.fold_spectra is something I should talk about with phil or you.

Reply:
Ah, this response should be for your comment on get_stats, not get_cov. My mistake, and I guess the comment wasn't mine oops.